### ISSUE TITLE
Can't use emojis in table names

### ISSUE THREAD

**Issue by snickell** [code-dot-org/code-dot-org#56766]
original: From Jessie: Random UI bug: I created a table with a couple columns. Then I created a new table with an invalid table name of “😍😂😋” that opened a preview of a new table with that name even though that table wasn’t actually created. Strangely, it also had the column headers from the previous table. On top of that, when I try to add a row to that table, it hangs. Then, when I switch to the original table, the row I added is there, still hanging.

Firebase lets you use emojis in the name, no problem. So there might be tables with names like this that we have to port over. 

The error we get is:

ActiveRecord::StatementInvalid in DatablockStorageController#create_table

Mysql2::Error: Illegal mix of collations (utf8_unicode_ci,IMPLICIT) and (utf8mb4_unicode_ci,COERCIBLE) for operation '='

app/controllers/datablock_storage_controller.rb:278:in `table_or_create'
app/controllers/datablock_storage_controller.rb:64:in `create_table'

**Comment by snickell** [comment]
I believe this is the line that's failing:
```
  def where_table
    DatablockStorageTable.where(project_id: @project_id, table_name: params[:table_name])
  end
```

Table on the adhoc looks like:
![Image](https://github.com/code-dot-org/code-dot-org/assets/223277/97b39766-6be1-4a39-a416-2a06452ac8cf)

**Comment by snickell** [comment]
Can repro with:
```
DatablockStorageTable.where(project_id: 5, table_name: "🎉").first_or_create

  DatablockStorageTable Load (0.7ms)  SELECT `datablock_storage_tables`.* FROM `datablock_storage_tables` WHERE `datablock_storage_tables`.`project_id` = 5 AND `datablock_storage_tables`.`table_name` = '🎉' ORDER BY `datablock_storage_tables`.`project_id` ASC, `datablock_storage_tables`.`table_name` ASC LIMIT 1
/Users/seth/.rbenv/versions/3.0.5/lib/ruby/gems/3.0.0/gems/mysql2-0.5.5/lib/mysql2/client.rb:151:in `_query': Mysql2::Error: Illegal mix of collations (utf8_unicode_ci,IMPLICIT) and (utf8mb4_unicode_ci,COERCIBLE) for operation '=' (ActiveRecord::StatementInvalid)
/Users/seth/.rbenv/versions/3.0.5/lib/ruby/gems/3.0.0/gems/mysql2-0.5.5/lib/mysql2/client.rb:151:in `_query': Illegal mix of collations (utf8_unicode_ci,IMPLICIT) and (utf8mb4_unicode_ci,COERCIBLE) for operation '=' (Mysql2::Error)
```

**Comment by snickell** [comment]
Similar issue for the value of KVPs:
```
(DatablockStorageKvp.new value: { name: "Seth"}, key: "Hi 👋").save!

  TRANSACTION (0.3ms)  BEGIN
  DatablockStorageKvp Create (0.6ms)  INSERT INTO `datablock_storage_kvps` (`key`, `value`) VALUES ('Hi 👋', '{\"name\":\"Seth\"}')
  TRANSACTION (0.1ms)  ROLLBACK
/Users/seth/.rbenv/versions/3.0.5/lib/ruby/gems/3.0.0/gems/mysql2-0.5.5/lib/mysql2/client.rb:151:in `_query': Mysql2::Error: Incorrect string value: '\xF0\x9F\x91\x8B' for column 'key' at row 1 (ActiveRecord::StatementInvalid)
/Users/seth/.rbenv/versions/3.0.5/lib/ruby/gems/3.0.0/gems/mysql2-0.5.5/lib/mysql2/client.rb:151:in `_query': Incorrect string value: '\xF0\x9F\x91\x8B' for column 'key' at row 1 (Mysql2::Error)
```

**Comment by snickell** [comment]
Fixed by #56837
